graphql_compiler/compiler/expressions.py

```diff
@@ -732,15 +734,27 @@ def to_sql(self, aliases, current_alias):
                 inner_type
             ))
 
-        # coalesce to an empty array of the corresponding type
-        empty_array = 'ARRAY[]::{}[]'.format(sql_array_type)
-        return sqlalchemy.func.coalesce(
-            aliases[
-                self.fold_scope_location.base_location.query_path,
-                self.fold_scope_location.fold_path
-            ].c['fold_output_' + self.fold_scope_location.field],
-            sqlalchemy.literal_column(empty_array)
-        )
+        fold_output_column = aliases[
+            self.fold_scope_location.base_location.query_path,
+            self.fold_scope_location.fold_path
+        ].c['fold_output_' + self.fold_scope_location.field]
+
+        if isinstance(dialect, MSDialect_pyodbc):
```

Comment:
MSDialect_pyodbc is the type of the SQLAlchemySchemaInfo compiler. In theory it would be great if we could do something like `dialect == sqlalchemy.dialects.mssql()` but different calls to the compiler generator don't return equal compilers.

Reply:
I think what you want to do here is:
```
from sqlalchemy.dialects.mssql.base import MSDialect
if isinstance(dialect, MSDialect)
```
`MSDialect_pyodbc` is a subclass of `MSDialect`

Reply:
Good idea